packages/blade/src/components/Input/PasswordInput/PasswordInput.tsx

```diff
@@ -67,30 +68,34 @@ type PasswordInputProps = Pick<
 > &
   PasswordInputExtraProps;
 
-const PasswordInput = ({
-  label,
-  labelPosition = 'top',
-  showRevealButton = true,
-  maxCharacters,
-  validationState,
-  errorText,
-  successText,
-  helpText,
-  isDisabled = false,
-  defaultValue,
-  placeholder,
-  isRequired = false,
-  necessityIndicator = 'none',
-  value,
-  onChange,
-  onFocus,
-  onBlur,
-  name,
-  autoFocus = false,
-  keyboardReturnKeyType = 'done',
-  autoCompleteSuggestionType,
-}: PasswordInputProps): ReactElement => {
-  const [isRevealed, setIsRevealed] = useState(false);
+const _PasswordInput: React.ForwardRefRenderFunction<BladeElementRef, PasswordInputProps> = (
+  {
+    label,
+    labelPosition = 'top',
+    showRevealButton = true,
+    maxCharacters,
+    validationState,
+    errorText,
+    successText,
+    helpText,
+    isDisabled = false,
+    defaultValue,
+    placeholder,
+    isRequired = false,
+    necessityIndicator = 'none',
+    value,
+    onChange,
+    onFocus,
+    onBlur,
+    name,
+    autoFocus = false,
+    keyboardReturnKeyType = 'done',
+    autoCompleteSuggestionType,
+  },
+  ref,
+) => {
+  const inputRef = useBladeInnerRef(ref);
```

Comment:
Yeah we can do that. at this point we are doing it on all the other components.

Reply:
Okay wait it's actually simpler to do this in here only. 

Because, for example take TextInput: 

We need the ref access in the TextInput itself to handle `textInputRef.current.clear()` & `textInputRef.current.focus()` for the clear button, plus we also need to pass down ref to the BaseInput to be able to populate the ref with the DOM element. 

If we remove useBladeInnerRef() here and move it to BaseInput, then in TextInput we need to merge the incoming ref & the textInputRef: 

````ts
const _TextInput = (
  props,
  incomingRef,
) => {
  const textInputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
  ...
  ...
  const renderInteractionElement = () => {
    // here we need access to textInputRef
    // textInputRef.current.focus() / clear()
  }

  return <BaseInput ref={mergeRefs([incomingRef, textInputRef]} /> // need to merge ref here
}
````

Reply:
alright this makes sense now. lets not change it then